src/tokens/TokenImport.tsx

```diff
@@ -0,0 +1,195 @@
+import React, { useState } from 'react'
+import { useTranslation } from 'react-i18next'
+import { ScrollView, StyleSheet, Text, View } from 'react-native'
+import { SafeAreaView } from 'react-native-safe-area-context'
+import { useDispatch } from 'react-redux'
+import { showMessage } from 'src/alert/actions'
+import { AssetsEvents } from 'src/analytics/Events'
+import ValoraAnalytics from 'src/analytics/ValoraAnalytics'
+import BackButton from 'src/components/BackButton'
+import Button, { BtnSizes } from 'src/components/Button'
+import InLineNotification, { Severity } from 'src/components/InLineNotification'
+import TextInput from 'src/components/TextInput'
+import CustomHeader from 'src/components/header/CustomHeader'
+import GreenLoadingSpinner from 'src/icons/GreenLoadingSpinner'
+import { noHeader } from 'src/navigator/Headers'
+import { navigateBack } from 'src/navigator/NavigationService'
+import { Screens } from 'src/navigator/Screens'
+import { StackParamList } from 'src/navigator/types'
+import { NETWORK_NAMES } from 'src/shared/conts'
+import { Colors } from 'src/styles/colors'
+import { typeScale } from 'src/styles/fonts'
+import { Spacing } from 'src/styles/styles'
+import variables from 'src/styles/variables'
+import { PasteButton } from 'src/tokens/PasteButton'
+import networkConfig from 'src/web3/networkConfig'
+
+type Props = NativeStackScreenProps<StackParamList, Screens.TokenImport>
+
+export default function TokenImportScreen(_: Props) {
+  const { t } = useTranslation()
+  const dispatch = useDispatch()
+
+  const [tokenAddress, setTokenAddress] = useState('')
+  const [tokenSymbol, setTokenSymbol] = useState('')
+  const [networkName, setNetworkName] = useState(NETWORK_NAMES[networkConfig.defaultNetworkId])
+
+  const navigateBackAndToast = () => {
+    ValoraAnalytics.track(AssetsEvents.import_token_submit, {
+      tokenAddress,
+      tokenSymbol,
+      networkId: networkConfig.networkId,
+    })
+    navigateBack()
+    // TODO RET-891: do this only when actually imported
+    dispatch(showMessage(t('tokenImport.importSuccess', { tokenSymbol })))
+  }
+
+  const errorMessage = () => {
+    // TODO RET-892: when states and validation are added, choose appropriate error or return null
+    const errors = [
+      t('tokenImport.error.invalidToken'),
+      t('tokenImport.error.alreadySupported'),
+      t('tokenImport.error.alreadyImported'),
+    ]
+    return <Text style={styles.errorLabel}>{errors[0]}</Text>
+  }
+
+  return (
+    <SafeAreaView style={styles.container}>
+      <CustomHeader
+        style={styles.customHeader}
+        left={<BackButton />}
+        title={<Text style={styles.headerTitle}>{t('tokenImport.title')}</Text>}
+      />
+      <ScrollView>
+        <View style={styles.titleContainer}>
+          <InLineNotification
+            severity={Severity.Informational}
+            description={t('tokenImport.notification')}
+          />
+
+          <View style={styles.inputContainer}>
```

Comment:
if you apply the gap style to the ScrollView instead, do you still need this View?

Reply:
The gaps differ between the notification and the container of the input groups. Otherwise we could remove this `<View>`. I'll check with Nitya if the gaps should be equal instead. For now I'll keep this `<View>`.

![image](https://github.com/valora-inc/wallet/assets/147835134/d1faa09a-6e40-4615-8de2-0379ba386ad2)